addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -17,6 +17,10 @@ const PopupWidget = publicWidget.Widget.extend({
      * @override
      */
     start: function () {
+        const hash = window.location.hash;
+        if (hash && this.$target.find('.modal').is('#' + $.escapeSelector(hash.substring(1)))) {
```

Comment:
That seems a bit excessive in order to merely check if the modal's `id` is `hash.substring(1)`.

Reply:
Also one concern here is that the fragment could be a lone `#` which I don't think would lead to a very useful behaviour.

Reply:
Hmmm that's unexpected and weird behaviour but you seem to be right, ignore that bit then. I assumed hash behaved in a regular manner without checking whether it did so, sorry about that.